### PR TITLE
cleanup Constants

### LINKED ISSUE #365: Get rid of outdated functions in Constants.*
The following functions are not really useful anymore:

https://github.com/precice/precice/blob/e480aef4e3a74ccd405f03a2770bee18cdb548e9/src/precice/Constants.hpp#L9-L13

We can get rid of them. Don't forget to also remove them in the bindings.

### PR DESCRIPTION
**Pull request by BenjaminRueth**
working on #365 here.

We drop the following functions from the API:

```
const std::string& nameConfiguration();
const std::string& dataDisplacements();
const std::string& dataForces();
const std::string& dataVelocities();
```

closes #365 

- [x] ~~check whether removal of `if ( context.triggerSolverPlot ) {...}` is possible. See https://github.com/precice/precice/pull/395/commits/103c8de46c500e4c2c43192f0f05915dc7aece38~~ #407 
- [x] remove functions from python language bindings
- [x] remove functions from c language bindings
- [x] remove functions from fortran language bindings

### CODE REVIEW

File: src/precice/bindings/c/SolverInterfaceC.h [hunk 1]
```diff
@@ -242,6 +242,8 @@ void precicec_mapWriteDataFrom ( int fromMeshID );
  */
 void precicec_mapReadDataTo ( int toMeshID );
 
+const char* precicec_actionWriteIterationCheckpoint();
+const char* precicec_actionReadIterationCheckpoint();
```
Comment (BenjaminRueth) [inline src/precice/bindings/c/SolverInterfaceC.h]: It looks like the following functions are missing in the C interface:
```
const std::string& actionWriteInitialData();
const std::string& actionPlotOutput();
int exportVTK();
int exportAll();
```

Comment (BenjaminRueth) [reply src/precice/bindings/c/SolverInterfaceC.h]: If we ignore

```
const std::string& actionPlotOutput();
int exportVTK();
int exportAll();
```

only 

```
const std::string& actionWriteInitialData();
```

is missing.

File: src/precice/impl/SolverInterfaceImpl.cpp [hunk 2]
```diff
@@ -439,7 +439,7 @@ void SolverInterfaceImpl:: finalize()
         suffix << _accessorName << ".final";
         exportMesh ( suffix.str() );
         if ( context.triggerSolverPlot ) {
```
Comment (BenjaminRueth) [inline src/precice/impl/SolverInterfaceImpl.cpp]: We can also think about entirely removing this block:

```
if ( context.triggerSolverPlot ) {
...
}
```

**Don't forget to check whether `context.triggerSolverPlot` and corresponding API can be removed**

File: src/precice/impl/SolverInterfaceImpl.cpp [hunk 3]
```diff
@@ -1537,7 +1537,7 @@ void SolverInterfaceImpl:: handleExports()
           suffix << _accessorName << ".dt" << _couplingScheme->getTimesteps()-1;
           exportMesh(suffix.str());
           if (context.triggerSolverPlot){
```
Comment (BenjaminRueth) [inline src/precice/impl/SolverInterfaceImpl.cpp]: We can also think about entirely removing this block:

```
if ( context.triggerSolverPlot ) {
...
}
```

**Don't forget to check whether `context.triggerSolverPlot` and corresponding API can be removed**

File: src/precice/impl/SolverInterfaceImpl.cpp [hunk 4]
```diff
@@ -318,7 +318,7 @@ void SolverInterfaceImpl:: initializeData ()
         suffix << _accessorName << ".init";
         exportMesh(suffix.str());
         if (context.triggerSolverPlot){
```
Comment (BenjaminRueth) [inline src/precice/impl/SolverInterfaceImpl.cpp]: We can also think about entirely removing this block:

```
if ( context.triggerSolverPlot ) {
...
}
```

**Don't forget to check whether `context.triggerSolverPlot` and corresponding API can be removed**

**Comment by BenjaminRueth** [comment]
Start review after https://github.com/precice/precice/pull/402 has been merged to develop.

File: src/precice/SolverInterface.hpp [hunk 5]
```diff
@@ -803,4 +802,17 @@ class SolverInterface
   friend struct testing::WhiteboxAccessor;
 };
 
+namespace constants {
+
+// @brief TODO
```
Comment (uekerman) [inline src/precice/SolverInterface.hpp]: TODO?

File: src/precice/bindings/c/SolverInterfaceC.h (hunk 1 above)
Comment (uekerman) [reply src/precice/bindings/c/SolverInterfaceC.h]: @BenjaminRueth promised that he would add `actionWriteInitialData` in another PR :smirk:

File: src/precice/bindings/fortran/SolverInterfaceFASTEST.cpp [hunk 6]
```diff
@@ -1,6 +1,6 @@
 #include "SolverInterfaceFASTEST.hpp"
 #include "SolverInterfaceFortran.hpp"
-#include "precice/impl/SolverInterfaceImpl.hpp"
+#include "precice/impl/SolverInterfaceImpl.hpp"  // TODO why not #include "precice/SolverInterface.hpp"?
```
Comment (uekerman) [inline src/precice/bindings/fortran/SolverInterfaceFASTEST.cpp]:
```suggestion
#include "precice/impl/SolverInterfaceImpl.hpp"  // @todo why not #include "precice/SolverInterface.hpp"?
```

Comment (BenjaminRueth) [reply src/precice/bindings/fortran/SolverInterfaceFASTEST.cpp]: https://github.com/precice/precice/pull/395/commits/cf98270cd05917868c0a21a62ca631c801f66a20 adds documentation

File: src/precice/SolverInterface.hpp (hunk 5 above)
Comment (BenjaminRueth) [reply src/precice/SolverInterface.hpp]: cf98270 adds documentation

**Comment by BenjaminRueth** [comment]
Can be merged as soon as #402 is closed and merged.

**Pull request merged**